services/messaging/src/main/java/org/eclipse/hono/messaging/SenderFactory.java

```diff
@@ -34,12 +34,14 @@
      * @param address The target address to use for the sender.
      * @param qos The quality of service the sender should use.
      * @param sendQueueDrainHandler The handler to notify about credits the sender is getting replenished with.
+     * @param closeHook The handler to invoke when the link is closed by the peer (may be {@code null}).
```

Comment:
if we keep the `Handler<String>` type then the JavaDoc should explain the meaning of the string that the handler will be invoked with, otherwise I propose to simply use `Handler<Void>`.

Reply:
Yes, I think the string does not help.